### PR TITLE
CLI: command to set deployment hooks on deployment configs

### PR DESCRIPTION
**Pull request by csrwng**
'oc set deployment-hook' command
Implements https://github.com/openshift/origin/issues/9040

### CODE REVIEW

File: pkg/cmd/cli/cmd/set/deploymenthook.go [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+
+	deploymentHookExample = `  # Clear pre and post hooks on a deployment config
+  %[1]s deployment-hook dc/myapp --remove --pre --post
+
+  # Set the pre deployment hook to execute a db migration command for an application
+  # using the data volume from the application
+  %[1]s deployment-hook dc/myapp --pre -v data -- /var/lib/migrate-db.sh
+
+  # Set a mid deployment hook along with additional environment variables
+  %[1]s deployment-hook dc/myapp --mid -v data -e VAR1=value1,VAR2=value2 -- /var/lib/prepare-deploy.sh`
```
Comment (bparees) [inline pkg/cmd/cli/cmd/set/deploymenthook.go]: i thought we were moving away from comma separated env variables and moving to mulitple -e args instead?

File: pkg/cmd/cli/cmd/set/deploymenthook.go [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+	cmd := &cobra.Command{
+		Use:     "deployment-hook DEPLOYMENTCONFIG --pre|--post|--mid -- CMD",
+		Short:   "Update a deployment hook on a deployment config",
+		Long:    deploymentHookLong,
+		Example: fmt.Sprintf(deploymentHookExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(options.Complete(f, cmd, args))
+			kcmdutil.CheckErr(options.Validate())
+			if err := options.Run(); err != nil {
+				// TODO: move met to kcmdutil
```
Comment (bparees) [inline pkg/cmd/cli/cmd/set/deploymenthook.go]: met?

**Comment by csrwng** [comment]
updated

**Comment by csrwng** [comment]
[test]

**Comment by csrwng** [comment]
@openshift/api-review

**Comment by bparees** [comment]
lgtm [merge]

**Pull request merged**